agent/lib_aws_sdk_php.c

```diff
@@ -295,6 +302,185 @@ void nr_lib_aws_sdk_php_sqs_parse_queueurl(
+    if (nr_php_is_zval_valid_integer(status_code)) {
+      external_params.status = Z_LVAL_P(status_code);
+    }
+    zval* metadata = nr_php_zend_hash_find(Z_ARRVAL_P(data), "@metadata");
+    if (NULL != metadata  && IS_REFERENCE == Z_TYPE_P(metadata)) {
+      metadata = Z_REFVAL_P(metadata);
+      if (IS_ARRAY == Z_TYPE_P(metadata)) {
+        zval* uri = nr_php_zend_hash_find(Z_ARRVAL_P(metadata), "effectiveUri");
+        if (nr_php_is_zval_valid_string(uri)) {
+          external_params.uri = Z_STRVAL_P(uri);
+        }
+      }
+    }
+  }
+  nr_segment_external_end(&auto_segment, &external_params);
+  nr_free(cloud_attrs.cloud_resource_id);
+}
+
+/* This stores the compiled regex to parse AWS ARNs. The compilation happens when
+ * it is first needed and is destroyed in mshutdown
+ */
+static nr_regex_t* aws_arn_regex;
+
+static void nr_aws_sdk_compile_regex(void) {
+  aws_arn_regex = nr_regex_create(AWS_ARN_REGEX, 0, 0);
+}
+
+void nr_aws_sdk_mshutdown(void) {
+  nr_regex_destroy(&aws_arn_regex);
+}
+
+void nr_aws_lambda_invoke(NR_EXECUTE_PROTO, nr_segment_cloud_attrs_t* cloud_attrs) {
+  zval* call_args = nr_php_get_user_func_arg(2, NR_EXECUTE_ORIG_ARGS);
+  zval* this_obj = NR_PHP_USER_FN_THIS();
+  char* arn = NULL;
+  char* function_name = NULL;
+  char* region = NULL;
+  zval* region_zval = NULL;
+  char* qualifier = NULL;
+  char* accountID = NULL;
+  bool using_account_id_ini = false;
+
+  /* verify arguments */
+  if (!nr_php_is_zval_valid_array(call_args)) {
+    return;
+  }
+  zval* lambda_args = nr_php_zend_hash_index_find(Z_ARRVAL_P(call_args), 0);
+  if (!nr_php_is_zval_valid_array(lambda_args)) {
+    return;
+  }
+  zval* lambda_name = nr_php_zend_hash_find(Z_ARRVAL_P(lambda_args), "FunctionName");
+  if (!nr_php_is_zval_valid_string(lambda_name)) {
+    return;
+  }
+
+  /* Ensure regex exists */
+  if (NULL == aws_arn_regex) {
+    nr_aws_sdk_compile_regex();
+  }
+
+  /* Extract all information possible from the passed lambda name via regex */
+  nr_regex_substrings_t* matches =
+      nr_regex_match_capture(aws_arn_regex,
+                             Z_STRVAL_P(lambda_name),
+                             Z_STRLEN_P(lambda_name));
+  function_name = nr_regex_substrings_get_named(matches, "functionName");
+  accountID = nr_regex_substrings_get_named(matches, "accountId");
+  region = nr_regex_substrings_get_named(matches, "region");
+  qualifier = nr_regex_substrings_get_named(matches, "qualifier");
+
+  /* suppliment missing information with API calls */
+  if (nr_strempty(function_name)) {
+    /*
+     * Cannot get the needed data. Function name is required in the
+     * argument, so this won't happen in normal operation
+     */
+    nr_free(function_name);
+    nr_free(accountID);
+    nr_free(region);
+    nr_free(qualifier);
+    nr_regex_substrings_destroy(&matches);
+    return;
+  }
+  if (nr_strempty(accountID)) {
+    nr_free(accountID);
+    accountID = NRINI(aws_account_id);
```

Comment:
What is the value of `NRINI(aws_account_id)` if `newrelic.cloud.aws.account_id` is not set? Empty string (`""`) or `NULL`?

Reply:
Default is set [here](https://github.com/newrelic/newrelic-php-agent/pull/1023/files#diff-b5410abe33a951704c8600c7b0b214d3c431cee94dcb3089567ec0ec9f328fc1R460), and should be set to empty string as default.
The code does check for empty/null here before trying to use it: https://github.com/newrelic/newrelic-php-agent/pull/1023/files#diff-b5410abe33a951704c8600c7b0b214d3c431cee94dcb3089567ec0ec9f328fc1R460